shell/common/shell_test.cc

```diff
@@ -19,7 +19,13 @@ namespace flutter {
 namespace testing {
 
 ShellTest::ShellTest()
-    : native_resolver_(std::make_shared<TestDartNativeResolver>()) {}
+    : native_resolver_(std::make_shared<TestDartNativeResolver>()),
+      assets_dir_(fml::OpenDirectory(GetFixturesPath(),
+                                     false,
+                                     fml::FilePermission::kRead)),
+      thread_host_("io.flutter.test." + GetCurrentTestName() + ".",
```

Comment:
I'm curious what's the difference between initialize `thread_host_` here and initialize `thread_host_` in `SetUp`, and what makes you change the old way.

Reply:
GTest allows to do the initialization in the constructor because a new instance of the test class is created for each test case.  So each test proceeds ctor() -> SetUp() -> test body -> TearDown() -> dtor().  In particular SetUp and TearDown are provided so that virtuals can be called before/after the test body.

Doing the initialization in the constructor has a few advantages:
1.  Member variables without a default() ctor can be initialized without relying on std::unique_ptr or std::optional
2.  Often the "TearDown" half of the code isn't even needed because the default destructor is sufficient.
3.  The overriding class can never forget to initialize like they can forget to call Base::Setup.

Doing the initialization in SetUp has a few advantages too:
1.  If you need to call a virtual during initialization, you have no choice but to do it here.
2.  Isn't as strict about the order you initialize things in, which can make the code easier to read.

We don't need to call a virtual here, so in the case of ThreadTest and friends I opted to change it, so that SkiaGpuObjectLayerTest doesn't need to override SetUp() or TearDown().

In the specific case of shell_test / ThreadHost, it looks like it has a default ctor + move constructor, so the unique_ptr was not necessary anyways :D  This code could be moved back to SetUp, but having it here makes the class easier to derive from IMO; less work for clients